packages/connect/src/createScClient/ScProvider/ScProvider.ts

```diff
@@ -87,57 +85,23 @@ const CONNECTION_STATE_PINGER_INTERVAL = 2000
  * ```
  */
 export class ScProvider implements ProviderInterface {
-  #chainSpec: string
   readonly #coder: RpcCoder = new RpcCoder()
   readonly #eventemitter: EventEmitter = new EventEmitter()
   readonly #handlers: Record<string, RpcStateAwaiting> = {}
   readonly #subscriptions: Record<string, StateSubscription> = {}
   readonly #waitingForId: Record<string, JsonRpcResponse> = {}
-  #connectionStatePingerId: ReturnType<typeof setInterval> | null
+  #connectionStatePingerId: ReturnType<typeof setInterval> | null = null
   #isConnected = false
+  #getChain: (handler: JsonRpcCallback) => Promise<Chain>
   #chain: Chain | undefined = undefined
-  #parachainSpecs: string | undefined = undefined
 
   /*
    * How frequently to see if we have any peers
    */
   healthPingerInterval = CONNECTION_STATE_PINGER_INTERVAL
 
-  /**
-   * @param knownChain - the name of a supported chain ({@link WellKnownChains})
-   * @param parachainSpec - optional param of the parachain chainSpecs to connect to
-   * @param autoConnect - whether the ScProvider should eagerly connect while its being instantiated. Defaults to `true`
-   *
-   */
-  public constructor(
-    knownChain: WellKnownChains,
-    parachainSpec?: string,
-    autoConnect?: boolean,
-  )
-  /**
-   * @param chainSpec - a string with the spec of the chain
-   * @param parachainSpec - optional param of the parachain chainSpecs to connect to
-   * @param autoConnect - whether the ScProvider should eagerly connect while its being instantiated. Defaults to `true`
-   *
-   */
-  public constructor(
-    chainSpec: string,
-    parachainSpec?: string,
-    autoConnect?: boolean,
-  )
-  public constructor(
-    chainSpec: string,
-    parachain?: string,
-    autoConnect = true,
-  ) {
-    this.#chainSpec = chainSpec
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-    this.#connectionStatePingerId = null
-    if (parachain) {
-      this.#parachainSpecs = parachain
-    }
-    // eslint-disable-next-line @typescript-eslint/no-floating-promises
-    if (autoConnect) this.connect()
+  public constructor(getChain: (handler: JsonRpcCallback) => Promise<Chain>) {
```

Comment:
Would be great if this was not public

Reply:
yeah... I see what you are saying: despite the fact that we are not explicitly making this part of the public API, someone could still import it and use it via

```ts
import { ScProvider } from "@substrate/connect/dist/createScClient/ScProvider"
```

I really want to avoid that, so I will address that before mergin. Thanks for raising this!

Reply:
Actually, on a second thought, I will address this later as it's easier to do this as part of the refactor that I'm planning to do on the `ScProvider`.